csv_processor.bal

```diff
@@ -262,3 +261,58 @@ function createQuotationFromCSVData(string[][] inputCSVData) returns Quotation[]
     }
     return qutoations;
 }
+
+isolated function createSupplierFromCSVData(string[][] inputCsv) returns Supplier[]|error {
+    Supplier[] suppliers = [];
+    string[] errors = [];
+    foreach [int, string[]] [idx, line] in inputCsv.enumerate() {
+        int csvLineNo = idx + 1;
+        if line.length() != 4 {
+            return error(string `Invalid CSV Length in line:${csvLineNo}`);
+        }
+        var [name, shortName, email, phoneNumber] = readSupplierCsvLine(line);
+        boolean hasErrors = false;
+        if !isValidEmail(email) {
+            errors.push(string `Line:${csvLineNo}| provided email [${email}] is invalid`);
+            hasErrors = true;
+        }
+        phoneNumber = regex:replaceAll(phoneNumber, "\\s+", "");
+        if !isValidPhoneNumber(phoneNumber) {
+            errors.push(string `Line:${csvLineNo}| provided phone number [${phoneNumber}] is invalid`);
+            hasErrors = true;
+        }
+        if !hasErrors {
```

Comment:
can we use errors.lenght here, so we can get rid of hasErrors?

Reply:
but that will lead O(n) complexity for the check